openai_integration.py

```diff
@@ -0,0 +1,104 @@
+from dotenv import load_dotenv
+from datetime import datetime, timedelta
+import asyncio
+import openai
+import os
+
+# Load the environment variables from the .env file
+load_dotenv()
+
+# Counter for the number of requests made
+request_counter = 0
+
+# Last request time
+last_request_time = None
+
+# Number of requests allowed per time interval
+requests_per_interval = 3

```

Comment:
styling of constants is UPPER letters.
also, how did you decide there would be 3 requests per interval?